library_generation/utilities.py

```diff
@@ -1,125 +1,503 @@
+
+    # generate .OwlBot.yaml
+    yaml_file = ".OwlBot.yaml"
+    if not os.path.exists(f"{library_path}/{yaml_file}"):
+        __render(
+            template_name="owlbot.yaml.monorepo.j2",
+            output_name=f"{library_path}/{yaml_file}",
+            artifact_name=distribution_name_short,
+            proto_path=remove_version_from(proto_path),
+            module_name=repo_metadata["repo_short"],
+            api_shortname=library.api_shortname,
+        )
+
+    # generate owlbot.py
+    py_file = "owlbot.py"
+    if not os.path.exists(f"{library_path}/{py_file}"):
+        template_excludes = [
+            ".github/*",
+            ".kokoro/*",
+            "samples/*",
+            "CODE_OF_CONDUCT.md",
+            "CONTRIBUTING.md",
+            "LICENSE",
+            "SECURITY.md",
+            "java.header",
+            "license-checks.xml",
+            "renovate.json",
+            ".gitignore",
+        ]
+        __render(
+            template_name="owlbot.py.j2",
+            output_name=f"{library_path}/{py_file}",
+            should_include_templates=True,
+            template_excludes=template_excludes,
+        )
+
+
+def repo_level_post_process(
+    repository_path: str,
+    versions_file: str,
+) -> None:
+    """
+    Perform repository level post-processing
+    :param repository_path: the path of the repository
+    :param versions_file: the versions_txt contains version of modules
+    :return: None
+    """
+    print("Regenerating root pom.xml")
+    modules = __search_for_java_modules(repository_path)
+    __render(
+        template_name="root-pom.xml.j2",
+        output_name=f"{repository_path}/pom.xml",
+        modules=modules,
+    )
+    print("Regenerating gapic-libraries-bom")
+    bom_configs = __search_for_bom_artifact(repository_path)
+    monorepo_version = get_version_from(
+        versions_file=versions_file,
+        artifact_id="google-cloud-java",
+    )
+    __render(
+        template_name="gapic-libraries-bom.xml.j2",
+        output_name=f"{repository_path}/gapic-libraries-bom/pom.xml",
+        monorepo_version=monorepo_version,
+        bom_configs=bom_configs,
+    )
+
+
+def get_version_from(
+    versions_file: str, artifact_id: str, is_released: bool = False
+) -> str:
+    """
+    Get version of a given artifact from versions.txt
+    :param versions_file: the path of version.txt
+    :param artifact_id: the artifact id
+    :param is_released: whether returns the released or current version
+    :return: the version of the artifact
+    """
+    index = 1 if is_released else 2
+    with open(versions_file, "r") as f:
+        for line in f.readlines():
+            if artifact_id in line:
+                return line.split(":")[index].strip()
+
+
+def main(argv: Sequence[str]) -> None:
+    if len(argv) < 1:
+        raise ValueError(
+            "Usage: python generate_composed_library_args.py function_name arg1...argN"
```

Comment:
I think it should be `python utilities.py`
However this `main` function was made so shell scripts can call them. Since we are moving to ITs written in python, it may make sense to remove them

Reply:
Removed.